### PR TITLE
feat: add loss + wrapper for MASTER model

### PR DESCRIPTION
**Pull request by charlesmindee**
This PR implements compute_loss and master function wrapping the MASTER model. Added corresponding unitests.
Master is now available in the model zoo as a recognition predictor.

Any feedback is welcome!

### CODE REVIEW

File: doctr/models/recognition/master.py [hunk 1]
```diff
@@ -275,3 +353,79 @@ def decode(self, inputs: tf.Tensor) -> Tuple[tf.Tensor, tf.Tensor]:
...
+        load_pretrained_params(model, default_cfgs[arch]['url'])
+
+    return model
+
+
+def master(pretrained: bool = False, **kwargs: Any) -> MASTER:
+    """MASTER as described in paper: <https://arxiv.org/pdf/1910.02562.pdf>`_.
+    Example::
+        >>> import tensorflow as tf
+        >>> from doctr.models import sar_vgg16_bn
```
Comment (fg-mindee) [inline doctr/models/recognition/master.py:421]: wrong import

File: doctr/models/recognition/master.py [hunk 2]
```diff
@@ -197,56 +212,119 @@ def __init__(
...
-        return logits
+        else:
+            raw_predictions, logits = self.decode(encoded)
+            if return_model_output:
+                out['out_map'] = logits
+            if return_preds:
+                predictions = self.postprocessor(logits)
+                out['preds'] = predictions
+
+        return out
```
Comment (fg-mindee) [inline doctr/models/recognition/master.py:315]: Could you rewrite that so that you get the same structure as in CRNN or SAR?
Most conditions should be independent of the others

**Review by fg-mindee — COMMENTED** [review]
Thanks for the PR! A short rewrite of the call method would be better to get a similar structure to the one of other recognition's models

File: doctr/models/recognition/master.py [hunk 3]
```diff
@@ -275,3 +346,79 @@ def decode(self, inputs: tf.Tensor) -> Tuple[tf.Tensor, tf.Tensor]:
...
+    # Patch the config
+    _cfg = deepcopy(default_cfgs[arch])
+    _cfg['input_shape'] = input_shape or _cfg['input_shape']
+    _cfg['vocab'] = kwargs.get('vocab', _cfg['vocab'])
+    _cfg['d_model'] = kwargs.get('d_model', _cfg['d_model'])
+    _cfg['headers'] = kwargs.get('headers', _cfg['headers'])
+    _cfg['dff'] = kwargs.get('dff', _cfg['dff'])
+    _cfg['num_heads'] = kwargs.get('num_heads', _cfg['num_heads'])
+    _cfg['num_layers'] = kwargs.get('num_layers', _cfg['num_layers'])
+    _cfg['max_length'] = kwargs.get('max_length', _cfg['max_length'])
```
Comment (fg-mindee) [inline doctr/models/recognition/master.py:391]: All of those don't need to be in the config (at least for now)

**Review by fg-mindee — COMMENTED** [review]
Last comment and we're good to go!

**Review by fg-mindee — APPROVED** [review]
Cheers!
